### ISSUE TITLE
Neural-net-basics

### ISSUE THREAD

**Issue by souzaalvesguilherme3-droid** [souzaalvesguilherme3-droid/Guilherme-de-Souza-alves-#7]
# Importando bibliotecas necessárias
import tensorflow as tf
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense
import matplotlib.pyplot as plt

# Definindo os dados de treino (x e y)
X_train = tf.constant([[1.0], [2.0], [3.0], [4.0], [6.0], [8.0]])
y_train = tf.constant([[2.0], [4.0], [6.0], [8.0], [12.0], [16.0]])  # saída esperada

# Criando o modelo de regressão linear simples
modelo = Sequential()
modelo.add(Dense(units=1, input_shape=[1]))  # 1 neurônio, entrada de 1 dimensão
modelo.compile(optimizer='sgd', loss='mean_squared_error')  # descida do gradiente

# Treinando o modelo (200 épocas = número de vezes que o modelo “aprende”)
modelo.fit(X_train, y_train, epochs=200, verbose=0)

# Fazendo uma previsão para um novo valor
X_new = tf.constant([[5.0]])
prediction = modelo.predict(X_new)
print("Predição:", prediction[0][0])  # resultado previsto pelo modelo

# Visualizando os pontos de treino e a linha de regressão aprendida
plt.scatter(X_train, y_train, label='Dados de treino')
plt.plot(X_train, modelo.predict(X_train), color='red', label='Linha de regressão')
plt.xlabel('x')
plt.ylabel('y (notas)')
plt.legend()
plt.show()

](https://github.com/user-attachments/assets/0c8a4841-b436-4dee-8198-7aa7f468288b)
